wso2-jenkins-jacoco-plugin/src/main/java/hudson/plugins/jacoco/report/CoverageReport.java

```diff
@@ -1,114 +1,109 @@
+        try {
+
+            action.getLogger().println("[JaCoCo plugin] Loading packages..");
+
+            if (executionFileLoader.getBundleCoverage() != null) {
+                setAllCovTypes(this, executionFileLoader.getBundleCoverage());
+
+                ArrayList<IPackageCoverage> packageList = new ArrayList<>(executionFileLoader.getBundleCoverage().getPackages());
+                for (IPackageCoverage packageCov : packageList) {
+                    PackageReport packageReport = new PackageReport();
+                    packageReport.setName(packageCov.getName());
+                    packageReport.setParent(this);
+                    this.setCoverage(packageReport, packageCov);
+
+                    ArrayList<IClassCoverage> classList = new ArrayList<>(packageCov.getClasses());
+                    for (IClassCoverage classCov : classList) {
+                        ClassReport classReport = new ClassReport();
+                        classReport.setName(classCov.getName());
+                        classReport.setParent(packageReport);
                         classReport.setSrcFileInfo(classCov, executionFileLoader.getSrcDir() + "/" + packageCov.getName() + "/" + classCov.getSourceFileName());
 
-						packageReport.setCoverage(classReport, classCov);
-
-						ArrayList<IMethodCoverage> methodList = new ArrayList<>(classCov.getMethods());
-						for (IMethodCoverage methodCov: methodList) {
-							MethodReport methodReport = new MethodReport();
-							methodReport.setName(getMethodName(classCov, methodCov));
-							methodReport.setParent(classReport);
-							classReport.setCoverage(methodReport, methodCov);
-							methodReport.setSrcFileInfo(methodCov);
-
-							classReport.add(methodReport);
-						}
-
-						packageReport.add(classReport);
-					}
-
-					this.add(packageReport);
-				}
-			}
-			action.getLogger().println("[JaCoCo plugin] Done.");
-			
-		} catch (Exception e) {
-			e.printStackTrace();
-		}
-	}
+                        packageReport.setCoverage(classReport, classCov);
+
+                        ArrayList<IMethodCoverage> methodList = new ArrayList<>(classCov.getMethods());
+                        for (IMethodCoverage methodCov : methodList) {
+                            MethodReport methodReport = new MethodReport();
+                            methodReport.setName(getMethodName(classCov, methodCov));
+                            methodReport.setParent(classReport);
+                            classReport.setCoverage(methodReport, methodCov);
+                            methodReport.setSrcFileInfo(methodCov);
+
+                            classReport.add(methodReport);
+                        }
+
+                        packageReport.add(classReport);
+                    }
+
+                    this.add(packageReport);
+                }
+            }
+            action.getLogger().println("[JaCoCo plugin] Done.");
+
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
1. Unless there's a better reason, catching generic Exception is not a good idea.

2. Use of printStackTrace is considered a bad practice.

Reply:
This is original source code